### PR TITLE
Fix analyzer on update item

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/UpdateLaunchHandlerImpl.java [hunk 1]
```diff
@@ -162,7 +163,11 @@ public OperationCompletionRS startLaunchAnalyzer(ReportPortalUser.ProjectDetails
 		List<Long> itemIds = collectItemsByModes(project, user.getUsername(), launch.getId(), analyzeRQ.getAnalyzeItemsMode());
 
 		analyzerServiceAsync.analyze(launch, itemIds, analyzerConfig)
-				.thenApply(it -> logIndexer.indexItemsLogs(project.getId(), launch.getId(), itemIds, analyzerConfig));
+				.thenApply(it -> CompletableFuture.supplyAsync(() -> logIndexer.indexItemsLogs(project.getId(),
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/launch/impl/UpdateLaunchHandlerImpl.java]: i think we don't need indexing to be async here, cause it is already async for analyze

File: src/main/java/com/epam/ta/reportportal/core/events/handler/TestItemFinishedEventHandler.java [hunk 2]
```diff
@@ -53,11 +55,11 @@ public void onApplicationEvent(ItemFinishedEvent itemFinishedEvent) {
 
 		AnalyzerConfig analyzerConfig = AnalyzerUtils.getAnalyzerConfig(project);
 
-		logIndexer.indexItemsLogs(
+		CompletableFuture.supplyAsync(() -> logIndexer.indexItemsLogs(
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/TestItemFinishedEventHandler.java]: maybe we need the whole event async here that is executed after main transaction is commited

File: src/main/java/com/epam/ta/reportportal/core/events/handler/LaunchFinishedEventHandler.java [hunk 3]
```diff
@@ -127,16 +127,18 @@ public void onApplicationEvent(LaunchFinishedEvent event) throws ExecutionExcept
...
 		if (BooleanUtils.isTrue(analyzerConfig.getIsAutoAnalyzerEnabled()) && analyzerServiceAsync.hasAnalyzers()) {
 			List<Long> itemIds = analyzeCollectorFactory.getCollector(AnalyzeItemsMode.TO_INVESTIGATE)
 					.collectItems(project.getId(), launch.getId(), null);
-			analyzerServiceAsync.analyze(launch, itemIds, analyzerConfig)
-					.thenApply(it -> logIndexer.indexItemsLogs(project.getId(), launch.getId(), itemIds, analyzerConfig));
+			logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig).join();
+			analyzerServiceAsync.analyze(launch, itemIds, analyzerConfig).join();
+			CompletableFuture.supplyAsync(() -> logIndexer.indexItemsLogs(project.getId(), launch.getId(), itemIds, analyzerConfig));
+		} else {
+			logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/LaunchFinishedEventHandler.java]: check the if condition
you will start indexing even you don't have any analyzers

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/LaunchFinishedEventHandler.java]: logs should be indexed in any case. It doesn't depend on auto analysis

File: src/main/java/com/epam/ta/reportportal/core/events/handler/TestItemFinishedEventHandler.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/TestItemFinishedEventHandler.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/UpdateLaunchHandlerImpl.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/UpdateLaunchHandlerImpl.java]: fixed

**Pull request merged**